Sources/Host/POSIX/Platform.cpp

```diff
@@ -124,6 +127,23 @@ bool Platform::SetWorkingDirectory(std::string const &directory) {
 
 ds2::ProcessId Platform::GetCurrentProcessId() { return ::getpid(); }
 
+// Termiantes the process and waits for it to exit before returning.
+bool Platform::TerminateProcess(ProcessId pid) {
+  if (::kill(pid, SIGKILL) < 0)
```

Comment:
what if the pid already doesn't exist as potentially the user terminated manually for some reason. Does it make sense to check for `errno == ESRCH`?

Reply:
This is a good catch; `kill` will fail with `ESRCH` if the process has already exited even if it hasn't been `wait`ed on by the parent yet. I'll fix this so it calls `waitpid` in the case of `errno == ESRCH` to ensure the process gets cleaned-up.